command/agent/consul/unit_test.go

```diff
@@ -1369,6 +1369,39 @@ func TestConsul_CanaryTags(t *testing.T) {
 	require.Len(ctx.FakeConsul.services, 0)
 }
 
+// TestConsul_CanaryTags_NoTags asserts Tags are used when Canary=true and there
+// are no specified canary tags
+func TestConsul_CanaryTags_NoTags(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	ctx := setupFake(t)
+
+	tags := []string{"tag1", "foo"}
+	ctx.Task.Canary = true
+	ctx.Task.Services[0].Tags = tags
```

Comment:
would suggest editing the other test above this to set both tags and Canarytags.

Reply:
The generic test task has it set: https://github.com/hashicorp/nomad/blob/796e623ddece8446f7f17b60e4a8e1ae608a4d07/command/agent/consul/unit_test.go#L39